pytest/lib/mocknet.py

```diff
@@ -1193,59 +1245,83 @@ def start_genesis_updater(node, script, genesis_filename_in,
                           epoch_length, node_pks, increasing_stakes, num_seats,
                           single_shard, all_node_pks, node_ips, neard):
     logger.info(f'Starting genesis_updater on {node.instance_name}')
-    node.machine.run('bash',
-                     input=start_genesis_updater_script(
-                         script, genesis_filename_in, records_filename_in,
-                         config_filename_in, out_dir, chain_id, validator_keys,
-                         rpc_nodes, done_filename, epoch_length, node_pks,
-                         increasing_stakes, num_seats, single_shard,
-                         all_node_pks, node_ips, neard))
+    node.machine.run(
+        'bash',
+        input=start_genesis_updater_script(
+            script,
+            genesis_filename_in,
+            records_filename_in,
+            config_filename_in,
+            out_dir,
+            chain_id,
+            validator_keys,
+            rpc_nodes,
+            done_filename,
+            epoch_length,
+            node_pks,
+            increasing_stakes,
+            num_seats,
+            single_shard,
+            all_node_pks,
+            node_ips,
+            neard,
+        ),
+    )
 
 
-def start_genesis_update_waiter_script(done_filename):
-    return '''
-        rm /home/ubuntu/waiter.txt
-        until [ -f {done_filename} ]
-        do
-            echo 'waiting for {done_filename}' >> /home/ubuntu/waiter.txt
-            date >> /home/ubuntu/waiter.txt
-            ls {done_filename} >> /home/ubuntu/waiter.txt
-            sleep 5
-        done
-        echo 'File found' >> /home/ubuntu/waiter.txt
-    '''.format(dir=shlex.quote(PYTHON_DIR),
-               done_filename=shlex.quote(done_filename))
+def wait_genesis_updater_done(node, done_filename):
+    msg = f'Waiting for the genesis updater on {node.instance_name}'
+    logger.info(msg)
 
+    # Wait until the neard process stops running.
+    while True:
+        time.sleep(5)
+        result = node.machine.run('ps | grep neard')
+        if result.returncode != 0:
+            break
 
-def wait_genesis_updater_done(node, done_filename):
-    logger.info(f'Waiting for the genesis updater on {node.instance_name}')
-    node.machine.run('bash',
-                     input=start_genesis_update_waiter_script(done_filename))
-    logger.info(
-        f'Waiting for the genesis updater on {node.instance_name} -- done')
+    # Check if the done_filename was produced.
+    result = node.machine.run(f'ls {done_filename}')
+    if result.returncode != 0:
+        # Get the stderr of the genesis updater.
+        script = f'cat {shlex.quote(PYTHON_DIR)}/genesis_updater.err'
+        stderr = node.machine.run(script).stdout
```

Comment:
nit (not a big deal): what if this cmd fails? then maybe we want a different error message below